File: 02-frontend/react-library-app/src/models/PaymentHistory.ts
class PaymentHistory {
  id: number;                
  userEmail: string;         
  invoiceReference: string;  
  amount: number;            
  paymentDate: string;       
  methodType: string;        

  constructor(id: number, userEmail: string, invoiceReference: string, amount: number, paymentDate: string, methodType: string) {
    this.id = id;
    this.userEmail = userEmail;
    this.invoiceReference = invoiceReference;
    this.amount = amount;
    this.paymentDate = paymentDate;
    this.methodType = methodType;
  }
}

export default PaymentHistory;

File: 01-backend/spring-boot-library/src/main/java/com/libranova/spring_boot_library/model/PaymentHistory.java
package com.libranova.spring_boot_library.model;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "payment_history")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PaymentHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "invoice_reference")
    private String invoiceReference;

    @Column(name = "amount")
    private Double amount;

    @Column(name = "payment_date")
    private LocalDateTime paymentDate;

    @Column(name = "method_type")
    private String methodType;
}
